device.go

```diff
@@ -32,6 +32,7 @@ type device struct {
 	BootEnvReadWriter
 	Commander
 	*partitions
+	hasUpdateCb func(d *device) (bool, error)
```

Comment:
@bboozzoo the problem is that we need to read the env twice, and there is no way to mock env read/writer to return two different things for two different calls.